### PR TITLE
Added support for avro4s json serialization/deserialization library

### PR DESCRIPTION
**Pull request by dborisenko**

### CODE REVIEW

File: akka-http-avro4s/src/main/scala/de/heikoseeberger/akkahttpavro4s/AvroSupport.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  implicit def marshaller[A: SchemaFor: ToRecord]: ToEntityMarshaller[A] = {
+    def encode(data: A): String = {
+      val baos = new ByteArrayOutputStream()
+      try {
+        val output = AvroOutputStream.json[A](baos)
+        try {
+          output.write(data)
+        } finally {
+          // Ignoring closing exception
+          Try(output.close())
```
Comment (hseeberger) [inline akka-http-avro4s/src/main/scala/de/heikoseeberger/akkahttpavro4s/AvroSupport.scala]: Please don't use allocating `Try`, use `try`-`catch` instead. Also below.

Comment (dborisenko) [inline akka-http-avro4s/src/main/scala/de/heikoseeberger/akkahttpavro4s/AvroSupport.scala]: Yes, sure. Done

File: akka-http-avro4s/src/main/scala/de/heikoseeberger/akkahttpavro4s/AvroSupport.scala [hunk 2]
```diff
@@ -81,13 +81,17 @@ trait AvroSupport {
         try {
           output.write(data)
         } finally {
-          // Ignoring closing exception
-          Try(output.close())
+          try output.close()
+          catch {
+            case _: IOException => // Ignoring closing exceptions
```
Comment (hseeberger) [inline akka-http-avro4s/src/main/scala/de/heikoseeberger/akkahttpavro4s/AvroSupport.scala]: Thanks for the change. Now I have a question: is it safe to ignore the exception? I don't know Avro, so I might be totally wrong, but I could imagine `output.write` to just buffer and `output.close` to actually write, in which case the exception should not be ignored.

Comment (dborisenko) [inline akka-http-avro4s/src/main/scala/de/heikoseeberger/akkahttpavro4s/AvroSupport.scala]: Yes, you probably right. We shouldn't ignore the error, even in `close()`. I'll do an update.

**Pull request merged**